### PR TITLE
Capture output from metadata emulator during tests

### PR DESCRIPTION
**Pull request by evildour**

### CODE REVIEW

File: apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/MetadataFixtureBase.cs [hunk 1]
```diff
@@ -63,14 +63,30 @@ protected MetadataFixtureBase()
...
             emulatorProcess.EnableRaisingEvents = true;
             emulatorProcess.Exited += EmulatorProcess_Exited;
         }
 
         private void EmulatorProcess_Exited(object sender, EventArgs e)
         {
-            throw new InvalidOperationException("The metadata server emulator exited with code " + emulatorProcess.ExitCode);
+            var message = $"The metadata server emulator exited with code {emulatorProcess.ExitCode}";
+
+            var output = emulatorProcess.StandardOutput.ReadToEnd();
```
Comment (jskeet) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/MetadataFixtureBase.cs]: Waiting until the end until we read anything can end up with it hanging due to a full buffer.
Maybe use the ErrorDataReceived and OutputDataReceived events instead?

Comment (evildour) [inline apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/MetadataFixtureBase.cs]: Done

**Comment by evildour** [comment]
PTAL

**Comment by evildour** [comment]
As far as I can tell, yes. I did try to make sure I was getting both outputs in both cases: an early exit due to an error and a process kill when disposing the fixture. In the latter case I wasn't getting standard output, so that's why I added the flush call in the python file. If you think there's anything else I should test for, let me know.

**Comment by jskeet** [comment]
Nope, that's fine - merge away :)

**Pull request merged**